src/functions/schema.ts

```diff
@@ -59,33 +55,16 @@ export const schema: IFunction<ISchemaOptions> = (targetVal, opts, paths) => {
   const { schema: schemaObj } = opts;
 
   try {
-    if (!ajv.validate(schemaObj, targetVal) && ajv.errors) {
-      // TODO: potential performance improvements (compile, etc)?
-      const collectedErrors: string[] = [];
-
-      for (const error of ajv.errors) {
-        if (collectedErrors.length > 0) {
-          const index = collectedErrors.indexOf(error.keyword);
-          if (index !== -1) {
-            if (mergeErrors(results[index], error)) continue;
-          }
-        }
-
-        let message = error.message || '';
-
-        if (
-          error.keyword === 'additionalProperties' &&
-          (error.params as AJV.AdditionalPropertiesParams).additionalProperty
-        ) {
-          message += `: ${(error.params as AJV.AdditionalPropertiesParams).additionalProperty}`;
-        }
-
-        collectedErrors.push(error.keyword);
-        results.push({
-          path: [...path, ...formatPath(error.dataPath)],
-          message,
-        });
-      }
+    const validator = validators.get(schemaObj);
```

Comment:
Could we throw in some code comments that explain a tiny bit of "what" but mostly "why"? Both are a little unclear.

Reply:
@philsturgeon sure, added. The main reason behind making this change is a very old todo comment we used to have here. The comment was about using `ajv.compile` and this is exactly what the following PR implements.